### PR TITLE
Sharing my little bit of progress

### PR DESCRIPTION
**Pull request by haandol**
first of all, I figured out I don't need COCODetectionMetric because all I want to do which does not match COCO metric (such as increasing IOU and calculate its mAP stuff).
unfortunately, I made COCODetectionMetric work on my workload before I realize I don't need it.
so I share this code for your testing, and I hope it helped.
the whole code is lousy right now but that's enough to share for now, I think.

you tested it with yolo_finetune.py as src and remove batch_size from hpo. (batch_size for hpo occurs error and I'm working on it.)

### CODE REVIEW

**Comment by haandol** [comment]
you can simply run `python utils.py` to test converter

**Comment by athewsey** [comment]
Thanks for sharing @haandol ! I do like the idea of customizing a Gluon dataset class (or multiple classes) - it's a nice reusable interface and consistent with the [PipeModeDataset extension for TensorFlow](https://github.com/aws/sagemaker-tensorflow-extensions#sagemaker-pipe-mode).

The `feat/objdetcustom` branch now has a version of the code which should train successfully on the workshop dataset in Pipe Mode (and some extra text in the notebook talking about why it's set up like it is). It's still super messy and certainly at least a little inefficient, but might be worth checking out if you're still interested in getting Pipe Mode working for object detection!

It should be possible to refactor into a custom dataset class, but I'd need to investigate a little closer what level to make the override at and check the training procedure is OK with violating the __len__ interface promised all the way up in Gluon `Dataset`!
